File: app/src/main/java/chat/tox/antox/ShareActivity.java
package chat.tox.antox;

import android.app.SearchManager;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public class ShareActivity extends AppCompatActivity
{

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onNewIntent(Intent intent)
    {

        if (Intent.ACTION_SEARCH.equals(intent.getAction()))
        {
            String query = intent.getStringExtra(SearchManager.QUERY);
        }
        else
        {
            Uri data = intent.getData();
            String dataString = intent.getDataString();
            String shareWith = dataString.substring(dataString.lastIndexOf('/') + 1);
        }
    }
}

File: app/src/main/scala/chat/tox/antox/activities/ShareActivity.scala
package chat.tox.antox.activities

import android.app.SearchManager
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

class ShareActivity extends AppCompatActivity {
  override protected def onCreate(savedInstanceState: Bundle) {
    super.onCreate(savedInstanceState)
  }

  override protected def onNewIntent(intent: Intent) {
    if (Intent.ACTION_SEARCH == intent.getAction) {
      val query: String = intent.getStringExtra(SearchManager.QUERY)
    }
    else {
      val data: Uri = intent.getData
      val dataString: String = intent.getDataString
      val shareWith: String = dataString.substring(dataString.lastIndexOf('/') + 1)
    }
  }
}